models/avatar.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	"crypto/md5"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"code.gitea.io/gitea/modules/cache"
+	"code.gitea.io/gitea/modules/setting"
+)
+
+var ()
+
+// EmailHash represents a pre-generated hash map
+type EmailHash struct {
+	ID     int64  `xorm:"pk autoincr"`
+	Email  string `xorm:"UNIQUE NOT NULL"`
```

Comment:
Email length?

Reply:
We don't have any lengths for emails elsewhere...

```
// User represents the object of individual and member of organization.
type User struct {
	ID        int64  `xorm:"pk autoincr"`
	LowerName string `xorm:"UNIQUE NOT NULL"`
	Name      string `xorm:"UNIQUE NOT NULL"`
	FullName  string
	// Email is the primary email address (to be used for communication)
	Email                        string `xorm:"NOT NULL"`
...
```